src/execution_plan/ops/op_update.c

```diff
@@ -111,10 +117,22 @@ OpResult OpUpdateReset(OpBase *ctx) {
 
 /* Executes delayed updates. */
 void _UpdateEntities(OpUpdate *op) {
+    EntityUpdateCtx *entity_to_update;
     for(int i = 0; i < op->entities_to_update_count; i++) {
-        EntityProperty *dest_entity_prop = op->entities_to_update[i].dest_entity_prop;
-        SIValue new_value = op->entities_to_update[i].new_value;
-        dest_entity_prop->value = new_value;
+        entity_to_update = &op->entities_to_update[i];
+        // Retrieve variables from update context
+        Entity *entity = entity_to_update->entity_reference;
+        EntityProperty *property = &entity->properties[entity_to_update->prop_idx];
+        SIValue new_value = entity_to_update->new_value;
+
+        // Only worry about index updates for nodes right now
+        AST_GraphEntity *ge = entity_to_update->ge;
+        if (ge->t == N_ENTITY) {
+          LabelStore *store = NULL;
+          if (ge->label) store = GraphContext_GetStore(op->gc, ge->label, STORE_NODE);
+          GraphContext_UpdateNodeIndices(op->gc, store, entity->id, property, &new_value);
```

Comment:
When using SET clause to introduce new attributes, I guess it is OK to call GraphContext_UpdateNodeIndices, we'll simply realize that index does not exists...

Reply:
In the same way, we perform schema updates on all attributes instead of just new ones. I'm really looking forward to fixing up this operation, but let's just keep it working for the moment!